modules/Bio/EnsEMBL/Compara/RunnableDB/EpoLowCoverage/LowCoverageGenomeAlignment.pm

```diff
@@ -225,17 +232,119 @@ sub _write_output {
 	  #the left and right indexes.
 	  #	      $gata->store($genomic_align_tree, "skip_left_right_indexes");
 	  $gata->store($genomic_align_tree, $skip_left_right_index);
+	  $self->_assert_cigar_lines_in_block($genomic_align_tree->modern_genomic_align_block_id);
+	  $self->_assert_tree($genomic_align_tree);
 	  $self->_write_gerp_dataflow($genomic_align_tree->modern_genomic_align_block_id);
       }
-
-      #DO NOT COMMENT THIS OUT!!! (at least not permenantly). Needed
-      #to clean up after each job otherwise you get files left over from
-      #the previous job.
-      $self->cleanup_worker_temp_directory;
   
   return 1;
 }
 
+sub _assert_tree {
+    my ($self, $gat) = @_;
+
+    if ($gat->get_child_count == 3) {
+        # The only ternary nodes allowed are "root" nodes of unroot trees
+        # of 3 sequences
+        foreach my $child (@{$gat->children}) {
+            unless ($child->is_leaf) {
+                $self->_backup_data_and_throw($gat, sprintf("Node %s is the child of a ternay node, but node a leaf of the tree", $child->name));
+            }
+        }
+    } else {
+        # The tree must be binary
+        foreach my $node (@{$gat->get_all_nodes}) {
+            if ($node->get_child_count == 1) {
+                $self->_backup_data_and_throw($gat, sprintf("Node %s is unary. The tree has not been minimised", $node->name));
+            } elsif ($node->get_child_count >= 3) {
+                $self->_backup_data_and_throw($gat, sprintf("Node %s is not binary (%d children)", $node->name, $node->get_child_count));
+            }
+        }
+    }
+
+    # Only the leaves have aligned sequences, all from extant species
+    foreach my $node (@{$gat->get_all_nodes}) {
+        if ($node->is_leaf) {
+            if ($node->genomic_align_group->genome_db->name eq 'ancestral_sequences') {
+                $self->_backup_data_and_throw($gat, sprintf("Node %s is a leaf, but attached to ancestral_sequences", $node->name));
+            }
+        } else {
+            if ($node->genomic_align_group) {
+                $self->_backup_data_and_throw($gat, sprintf("Node %s is internal, but has some genomic_aligns", $node->name));
+            }
+        }
+    }
+}
+
+sub _assert_cigar_lines_in_block {
```

Comment:
I have considered moving this method in a generic form to `Utils/Cigars.pm` since you are using it in three different runnables, but it may be too complicated based on the many tailored bits (although the core is the same). Leaving it here just in case :)